src/client/clientmap.cpp

```diff
@@ -494,7 +494,7 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 	g_profiler->avg(prefix + "material swaps [#]", material_swaps);
 }
 
-static bool getVisibleBrightness(Map *map, const v3f &p0, v3f dir, float step,
+static bool getVisibleBrightness(Map *map, const v3opos_t &p0, v3f dir, float step,
```

Comment:
Why are you not switching the dir to doubles?

Reply:
because dir handle [0..1] values, why to switch?

Reply:
> because dir handle [0..1] values, why to switch?

Precision? No need for "ugly" casts? Consistency?